mitiq/interface/mitiq_pennylane/tests/test_conversions_pennylane.py

```diff
@@ -0,0 +1,123 @@
+    """Tests converting Pennylane expectation values."""
+    # <X>
+    with qml.tape.QuantumTape() as tape:
+        qml.expval(qml.PauliX(wires=[0]))
+
+    circuit = from_pennylane(tape)
+    q = cirq.LineQubit(0)
+    correct = cirq.Circuit(cirq.H.on(q))
+    assert _equal(circuit, correct, require_qubit_equality=False)
+
+    # <Z>
+    with qml.tape.QuantumTape() as tape:
+        qml.expval(qml.PauliZ(wires=[0]))
+
+    circuit = from_pennylane(tape)
+    correct = cirq.Circuit()
+    assert _equal(circuit, correct, require_qubit_equality=False)
+
+
+def test_from_pennylane():
+    with qml.tape.QuantumTape() as tape:
+        qml.CNOT(wires=[0, 1])
+
+    circuit = from_pennylane(tape)
+    correct = cirq.Circuit(cirq.CNOT(*cirq.LineQubit.range(2)))
+
+    assert _equal(circuit, correct, require_qubit_equality=False)
+
+
+def test_from_pennylane_unsupported_tapes():
+    with qml.tape.QuantumTape() as tape:
+        qml.CZ(wires=[0, "a"])
+
+    with pytest.raises(UnsupportedQuantumTapeError, match="could not sort"):
+        from_pennylane(tape)
+
+
+def test_no_variance():
+    with qml.tape.QuantumTape() as tape:
+        qml.CNOT(wires=[0, 1])
+        qml.var(qml.PauliZ(0))
+
+    with pytest.raises(
+        UnsupportedQuantumTapeError,
+        match="Only expectation value measurements",
+    ):
+        from_pennylane(tape)
+
+
+@pytest.mark.parametrize("random_state", range(10))
+def test_to_from_pennylane(random_state):
+    circuit = cirq.testing.random_circuit(
+        qubits=4, n_moments=2, op_density=1, random_state=random_state
+    )
+
+    converted = from_pennylane(to_pennylane(circuit))
+    # Gates (e.g. iSWAP) aren't guaranteed to be preserved. Check unitary
+    # instead of circuit equality.
+    cirq.testing.assert_allclose_up_to_global_phase(
+        cirq.unitary(converted), cirq.unitary(circuit), atol=1e-7
+    )
+
+
+def test_to_from_pennylane_cnot_same_gates():
+    qreg = cirq.LineQubit.range(2)
+    circuit = cirq.Circuit(cirq.CNOT(*qreg))
+    converted = from_pennylane(to_pennylane(circuit))
+    assert _equal(circuit, converted, require_qubit_equality=False)
+
+
+def test_to_from_pennylane_identity():
+    q = cirq.LineQubit(0)
+    # Empty circuit
+    circuit = cirq.Circuit()
+    converted = from_pennylane(to_pennylane(circuit))
+    assert _equal(circuit, converted, require_qubit_equality=False)
+    circuit = cirq.Circuit(cirq.I(q))
+    # Identity gate
+    converted = from_pennylane(to_pennylane(circuit))
+    # TODO: test circuit equality after Identity operation will be added
+    # to PennyLane (https://github.com/PennyLaneAI/pennylane/issues/1632)
+    assert np.allclose(cirq.unitary(circuit), cirq.unitary(converted))
+
+
+def test_non_consecutive_wires_error():
+    with qml.tape.QuantumTape() as tape:
+        qml.CNOT(wires=[0, 2])
+    with pytest.raises(
+        UnsupportedQuantumTapeError, match="contiguously pack",
+    ):
+        from_pennylane(tape)
```

Comment:
```suggestion
        from_pennylane(tape)


def test_integration():
    n_wires = 4

    gates = [
        qml.PauliX,
        qml.PauliY,
        qml.PauliZ,
        qml.S,
        qml.T,
        qml.RX,
        qml.RY,
        qml.RZ,
        qml.Hadamard,
        qml.Rot,
        qml.CRot,
        qml.Toffoli,
        qml.SWAP,
        qml.CSWAP,
        qml.U1,
        qml.U2,
        qml.U3,
        qml.CRX,
        qml.CRY,
        qml.CRZ,
    ]

    layers = 3
    np.random.seed(1967)
    gates_per_layers = [np.random.permutation(gates) for _ in range(layers)]

    with qml.tape.QuantumTape() as tape:
        np.random.seed(1967)
        for gates in gates_per_layers:
            for gate in gates:
                params = list(np.pi * np.random.rand(gate.num_params))
                rnd_wires = np.random.choice(range(n_wires), size=gate.num_wires, replace=False)
                gate(
                    *params,
                    wires=[
                        int(w) for w in rnd_wires
                    ]  # make sure we do not address wires as 0-d arrays
                )

    base_circ = from_pennylane(tape)
    tape_recovered = to_pennylane(base_circ)
    circ_recovered = from_pennylane(tape_recovered)

    u_1 = cirq.unitary(base_circ)
    u_2 = cirq.unitary(circ_recovered)

    assert np.allclose(u_1, u_2)

```
Proposing the above integration test, based upon [this](https://github.com/PennyLaneAI/pennylane/blob/032b3016919fd1671e26c6e89e46de967ed65836/pennylane/devices/tests/test_compare_default_qubit.py#L187) test.

Having this test pass makes me much more confident that reasonable circuits can be converted back and forth.